packages/wp-story-editor/src/api/story.js

```diff
@@ -111,6 +157,19 @@ export function saveStoryById(story, stories, encodeMarkup) {
     path,
     data: storySaveData,
     method: 'POST',
+  }).then((data) => {
+    const { _embedded: embedded = {} } = data;
+
+    if ('wp:featuredmedia' in embedded) {
+      data.featured_media = {
+        id: embedded?.['wp:featuredmedia']?.[0].id || 0,
+        height: embedded?.['wp:featuredmedia']?.[0]?.media_details?.height || 0,
+        width: embedded?.['wp:featuredmedia']?.[0]?.media_details?.width || 0,
+        url: embedded?.['wp:featuredmedia']?.[0]?.source_url || '',
+      };
+    }
```

Comment:
I'd remove the `if()` here. Then we don't need the `featuredMedia = featured_media ?? defaultMedia` check in `useLoadStory`

```suggestion
    data.featured_media = {
      id: embedded?.['wp:featuredmedia']?.[0].id || 0,
      height: embedded?.['wp:featuredmedia']?.[0]?.media_details?.height || 0,
      width: embedded?.['wp:featuredmedia']?.[0]?.media_details?.width || 0,
      url: embedded?.['wp:featuredmedia']?.[0]?.source_url || '',
    };
```